### PR TITLE
Add support for Samsung ARTIK boards (520, 710, 1020) and ARTIK SDK

### PR DESCRIPTION
**Pull request by glemercier**
First shot at ARTIK boards/SDK integration. Still need to push "framework-artik-sdk" package to a public repo.

### CODE REVIEW

File: platform.json [hunk 1]
```diff
@@ -33,6 +37,11 @@
       "type": "framework",
       "optional": true,
       "version": "~1.232.0"
+    },
+    "framework-artik-sdk": {
+      "type": "framework",
+      "optional": true,
+      "version": "1.0.2"
```
Comment (ivankravets) [inline platform.json:44]: Do you want to be fixed with strict version? How about `~1.0.2` which means `1.0.x`?

**Comment by ivankravets** [comment]
Thanks!

> Still need to push "framework-artik-sdk" package to a public repo.

How do you decide to do that? Would you like to have own repository manifest?

**Comment by glemercier** [comment]
Actually it seems I don't need a package for the SDK, since there is no specific files to install (I tested with a package containing only package.json). I will look into the "dnf" based installation check instead.

**Comment by ivankravets** [comment]
Is this SDK pre-installed by default on Artik? Or user should install it manually using `dnf` tool?

**Comment by ivankravets** [comment]
PIO Core has been updated and now allows framework without package:
https://github.com/platformio/platformio-core/commit/403da8e22b86e139fbb87ebe6bdddfd5b1bfbc6c

**Pull request merged**